packages/react-native-reanimated/src/platform-specific/RNRenderer.ts

```diff
@@ -1,4 +1,49 @@
+/* eslint-disable camelcase */
 // eslint-disable-next-line @typescript-eslint/ban-ts-comment
 // @ts-nocheck
 'use strict';
-export { default as RNRenderer } from 'react-native/Libraries/Renderer/shims/ReactNative';
+
+import { isFabric } from '../PlatformChecker';
+
+function findHostInstanceFastPath(ref: React.Component) {
+  if (ref.__internalInstanceHandle && ref.__nativeTag && ref._viewConfig) {
+    // This is a native ref to a Fabric component
+    return ref;
+  }
+  if (ref._nativeTag && ref.viewConfig) {
+    // This is a native ref to a Paper component
+    return ref;
+  }
+  return undefined;
```

Comment:
Shouldn't we throw an error in such case for easier debugging?

Reply:
When `findHostInstanceFastPath` returns `undefined` we call `findHostInstance_DEPRECATED()`

Reply:
This is not an error. That just means it’s a ref to a non-native component, and it’s necessary to call `findHostInstance_DEPRECATED` on them - I'll add a proper comment to code to explain it.